kong/cmd/utils/prefix_handler.lua

```diff
@@ -41,6 +41,21 @@ client:request { \
 %s -e "$CMD"
 ]]
 
+local stop_script_template = [[
+#!/bin/sh
+
+CMD="\
+local conf_loader = require 'kong.conf_loader' \
+local DAOFactory = require 'kong.dao.factory' \
+local Serf = require 'kong.serf' \
+local conf = assert(conf_loader('%s')) \
+local dao = DAOFactory(conf) \
+Serf.new(conf, dao):leave() \
+"
+
+%s -e "$CMD"
+]]
```

Comment:
@Tieske the script is being executed on forced shutdowns, if the database is down logs would be already printed in the regular `error.log` files because everything else would stop working. If the script for whatever reason fails, the node is purged anyways after `3600` seconds.